core/src/main/scala/kafka/zk/ZkData.scala

```diff
@@ -56,8 +57,8 @@ import scala.util.{Failure, Success, Try}
 
 object ControllerZNode {
   def path = "/controller"
-  def encode(brokerId: Int, timestamp: Long): Array[Byte] = {
-    Json.encodeAsBytes(Map("version" -> 1, "brokerid" -> brokerId, "timestamp" -> timestamp.toString).asJava)
+  def encode(brokerId: Int, timestamp: Long, isKRaft: Boolean = false): Array[Byte] = {
```

Comment:
One thing I was thinking is should we have the `kraftControllerEpoch` here instead of `isKraft`? Then `kraftControllerEpoch` == -1 (or not present) could mean ZK mode?

it might help with tracing what was going on...